src/classes/Player.ts

```diff
@@ -128,7 +128,7 @@ export class Player {
       if (objects[i].isInside('hitbox', this.position)) {
         this.interactingWithFragment = objects[i].isBeingInteractedWith();
         if(objects[i].isAllowedToInteract(myDirection)){
-          objects[i].interact(this, keyPressed);
+          objects[i].interact(keyPressed);
```

Comment:
Por que não precisa do this aqui mais?

Reply:
porque deixou de ser necessário passar o objeto do jogador, já que o estado do inventário vai ficar no singleton.